lib/gui/app/models/settings.js

```diff
@@ -156,9 +157,16 @@ exports.set = (key, value) => {
     }))
   }
 
+  const previousValue = settings[key]
+
   settings[key] = value
 
   return localSettings.writeAll(settings)
+    .catch((error) => {
```

Comment:
NIT: `writeAll` returns a Bluebird promise so you can use `.tapCatch`

Reply:
Aye, added!

Reply:
Had to revert this, as this method isn't a thing with the version of Bluebird we're using atm yet